### PR TITLE
fix: impoving transmission records by moving response body to new table

### PR DESCRIPTION
**Pull request by alex-sheehan-edx**
CAT-1 performance related issues

**IN THIS PR**
We are attempting to increase efficiency of the transmit content metadata task. In order of impact on the system performance we are-

**High Impact**
- Reducing the size each row in the `ContentMetadataItemTransmission` table by migrating it's largest field `api_response_body` to another table (`ApiResponseRecord`). Now, instead of saving the json blob to the transmission record, there is a 1:1 relationship between transmission records and the `ApiResponseRecord` table that houses the response information.
- Adding an index on the `ContentMetadataItemTransmission` table between the `enterprise_customer`,  `integrated_channel_code`, `plugin_configuration_id`, and `content_id` columns. Each task run does an enormous amount of table scans, querying by those fields. As such the index should help with the job run.

**Low Impact**
- slightly reduce the number of saves on the `ContentMetadataItemTransmission` during a content metadata transmission

**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.
- [ ] [Version](https://github.com/edx/edx-enterprise/blob/master/enterprise/__init__.py) bumped
- [ ] [Changelog](https://github.com/edx/edx-enterprise/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/edx/edx-enterprise/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [GitHub Actions](https://github.com/edx/edx-enterprise/actions), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-platform](https://github.com/edx/edx-platform) to upgrade dependencies (including edx-enterprise)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

File: integrated_channels/integrated_channel/exporters/content_metadata.py [hunk 1]
```diff
@@ -243,10 +243,10 @@ def _check_matched_content_to_create(
                         channel_metadata=None,
                         content_last_changed=content_last_changed,
                         enterprise_customer_catalog_uuid=enterprise_customer_catalog.uuid,
-                        plugin_configuration_id=self.enterprise_configuration.id
+                        plugin_configuration_id=self.enterprise_configuration.id,
```
Comment (alex-sheehan-edx) [inline integrated_channels/integrated_channel/exporters/content_metadata.py:246]: saving an additional table save

File: integrated_channels/integrated_channel/migrations/0021_remove_contentmetadataitemtransmission_api_response_body.py [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Generated by Django 3.2.15 on 2022-09-29 17:20
+
+from django.db import migrations
```
Comment (alex-sheehan-edx) [inline integrated_channels/integrated_channel/migrations/0021_remove_contentmetadataitemtransmission_api_response_body.py:3]: Removing a relatively large field from the table for performance reasons

File: integrated_channels/integrated_channel/migrations/0022_alter_contentmetadataitemtransmission_index_together.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# Generated by Django 3.2.15 on 2022-09-29 19:01
+
+from django.db import migrations
```
Comment (alex-sheehan-edx) [inline integrated_channels/integrated_channel/migrations/0022_alter_contentmetadataitemtransmission_index_together.py:3]: Adding an index on 
```
'enterprise_customer', 'integrated_channel_code', 'plugin_configuration_id', 'content_id'
```
to speed up table scans

File: integrated_channels/integrated_channel/models.py [hunk 4]
```diff
@@ -414,6 +439,8 @@ class ContentMetadataItemTransmission(TimeStampedModel):
 
     .. no_pii:
     """
+    class Meta:
```
Comment (alex-sheehan-edx) [inline integrated_channels/integrated_channel/models.py:442]: https://github.com/openedx/edx-enterprise/pull/1651/files#r983927867

**Review by johnnagro — APPROVED** [review]

File: integrated_channels/integrated_channel/transmitters/content_metadata.py [hunk 5]
```diff
@@ -184,7 +185,18 @@ def _transmit_action(self, content_metadata_item_map, client_method, action_name
                         course_or_course_run_key=content_id
                     )
                     transmission.api_response_status_code = response_status_code
```
Comment (bcitro) [inline integrated_channels/integrated_channel/transmitters/content_metadata.py:187]: should this line be deleted?

**Review by bcitro — APPROVED** [review]
One question otherwise LGTM

**Pull request merged**